test/src/accessibility.spec.ts

```diff
@@ -870,6 +890,28 @@ describe('Accessibility', function () {
         ).toEqual('<b>Hello, </b> world!');
       });
     });
+
+    it('should not report Document as leaf node', async () => {
+      const {page} = await getTestState();
+
+      await page.setContent(
+        html`<main><span>Hello</span><div> </div><div>World</div></main>`,
+      );
+
+      await page.setContent(html`<span>Name:</span><input />`);
```

Comment:
it has two setContents? cc @Lightning00Blade

Reply:
Ups I think we need only the second one

Reply:
I will send a fix

Reply:
Fixed in https://github.com/puppeteer/puppeteer/pull/14221